hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -99,6 +98,8 @@
     // to avoid reloading same items
     private long lastIdLoaded;
 
+    private volatile Map<Long, QueueItem> backupMap;
```

Comment:
The `backupMap` is initialized with the `HashMap` that is not thread-safe. Is it handled properly that multiple threads are not accessing it?

Reply:
fixed it by making `backupMap` a CHM.